src/tree_list/mod.rs

```diff
@@ -114,6 +123,14 @@ impl<T: Display + Debug> TreeList<T> {
         self.height = 0;
     }
 
+    pub fn first_col(&self, index: usize) -> Option<usize> {
+        self.items.get(index).and_then(|item| Some(item.offset()))
```

Comment:
Maybe you want `map` rather than `and_then` if the offset never fails?
Same for `width()`

Reply:
Good point, thanks.